<?xml version="1.0"?>
<launch>


  <param name="num_robots" value="4" type="int"/>
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="ip" default="vicon" />
  <arg name="port" default="3883" />
  <arg name="server" default="192.168.2.2"/>

  <arg name="uri1" default="radio://0/80/2M/E7E7E7E701" />
  <arg name="frame1" default="cf1" />


  <arg name="x1" default="0.5" />
  <arg name="y1" default="0.5" />
  <arg name="z1" default="1" />



  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="cf1">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri1)" />
      <param name="tf_prefix" value="cf1" />
      <param name="enable_logging" value="True" />
    </node>

<!--    <rosparam command="load" file="$(find crazyflie_demo)/launch/configuration/crazyflie_conf.yaml" />-->

    <node name="joy" pkg="joy" type="joy_node" output="screen" >
      <param name="dev" value="$(arg joy_dev)" />
    </node>


    <node name="joystick_controller" pkg="crazyflie_demo" type="controller.py" output="screen">
      <param name="use_crazyflie_controller" value="True" />
    </node>

    <include file="$(find crazyflie_controller)/launch/crazyflie2.launch">
      <arg name="frame" value="$(arg frame1)" />
    </include>

    <node name="pose" pkg="crazyflie_demo" type="publish_pose_teleop.py" output="screen">
      <param name="name" value="goal" />
      <param name="rate" value="30" />
      <param name="x" value="$(arg x1)" />
      <param name="y" value="$(arg y1)" />
      <param name="z" value="$(arg z1)" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0 0 0 0 1 $(arg frame1) /cf1/base_link 100" />

  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
    <rosparam subst_value="true">
      server: $(arg server)
      port: 3883

      update_frequency: 100.0
      frame_id: world

      # Use the VRPN server's time, or the client's ROS time.
      use_server_time: false
      broadcast_tf: true

      # Must either specify refresh frequency > 0.0, or a list of trackers to create
      refresh_tracker_frequency: 1.0
      #trackers:
      #- FirstTracker
      #- SecondTracker
    </rosparam>
  </node>
  </group>


  <!-- run rviz -->
<!--   <param name="robot_description" command="$(find xacro)/xacro.py $(find crazyflie_description)/urdf/crazyflie.urdf.xacro" />-->
<!--   <node name="rviz" pkg="rviz" type="rviz" args="-d $(find crazyflie_demo)/launch/crazyflie_pos.rviz" required="true" />-->

  <param name="robot_description" command="$(find xacro)/xacro.py $(find crazyflie_description)/urdf/crazyflie.urdf.xacro" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find crazyflie_demo)/launch/crazyflie_pos.rviz" required="true" />
</launch>


